yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -346,7 +347,16 @@ private[spark] class ApplicationMaster(
                 "Max number of executor failures reached")
             } else {
               logDebug("Sending progress")
-              allocator.allocateResources()
+              allocator.updateResources()
+
+              // Get the preemption executors from YarnAllocator, if preemption executor list is
+              // changed compared to last time, notify scheduler backend this preemption executor
+              // list.
+              val preemptionExecutors = allocator.getPreemptionExecutors
+              if (preemptionExecutors != lastPreemptionExecutors) {
+                amEndpoint.send(PreemptionExecutors(preemptionExecutors))
```

Comment:
I would incline to use `amEndpoint` here as to keep consistent with other places like `AddWebFIlter`.